numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -301,10 +301,10 @@ genint_type_str(PyObject *self)
             item = PyLong_FromUnsignedLong(*(uint32_t *)val);
```

Comment:
These and above are also wrong (or maybe just badly named).  Probably makes sense to just use `npy_<name>`, although they are the same anyway.
(I suppose, in practice we probably don't support any platform for which all but `long` aren't correct.)